### ISSUE TITLE
Redeployment with invalid Cron Schedule

### ISSUE THREAD

**Issue by BalduinB** [zeit/now#10894]
When a deployment with an invalid Cron-Job schedule is created, the deployment is not shown under "Deployments", not even as canceled.
When using the Vercel CLI to create a production build (`vercel --prod`) we get the following error:
```sh
vercel --prod
Vercel CLI 31.0.3
Error: Error while validating your Cron Jobs expressions:

            - Expected 5 values, but got 6. (Input cron: '10 11 * * * *')
```

The failed deployment should be visible on the website.

**Comment by BalduinB** [comment]
![Bildschirmfoto 2023-12-01 um 17 23 05](https://github.com/vercel/vercel/assets/43539017/33aaa650-646b-4d5d-8cbf-4e9e117ce652)

here is a screen shot from the GitHub integration, where the api has the invalid schedule format. 
As you can see there is also no `Details` button linking the deployment

**Comment by janwilmake** [comment]
I ran into this too. I was blind until I found this issue because I never use `Vercel --prod` and the error only showed up there: The error was `Error: Hobby accounts are limited to daily cron jobs. This cron expression (0 * * * *) would run more than once per day. Upgrade to pro to unlock all Cron Jobs features on Vercel.`

In GitHub it just said "deployment failed" with no further information.

This is not a good dev experience!
